### ISSUE TITLE
`match` with multiline array/dict pattern raises parse error

### ISSUE THREAD

**Issue by Scony** [godotengine/godot#85867]
### Tested versions

4.3.dev [d76c1d0e516fedc535a2e394ab780cac79203477]

### System information

Linux Manjaro

### Issue description

While it's totally fine to use array/dict pattern within `match` in a single line, it's not possible to spread the same code across multiple lines:
```
func foo(x):
	match x:
		[1, 2]:  # ok
			pass
		{"a": 1}:  # ok
			pass
		[  # Parse Error: Expected expression for match pattern.
			2,
			3
		]:
			pass
		{  # Parse Error: Expected expression as key for dictionary pattern.
			"a": 1
		}:
			pass
```
While line break (`\`) can be used as a workaround in that case, the above should be aligned to how expressions work i.e. it should be allowed to use multiline arrays/dicts.

### Steps to reproduce

See above

### Minimal reproduction project (MRP)

N/A

**Comment by Scony** [comment]
This is fixed with the latest `master`

**Comment by dalexeev** [comment]
* "Duplicate" of #90372.
* Fixed by #90373.
